hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,295 @@
+ * on the boundary and particle velocity is updated according to the boundary conditions.Streaming then continues and
+ * place the particle inside SphereGeometry. And particle which went outside or collided with geometry was marked.
+ * Right amount of marked particles are then evaporated.
+ *
+ * To facilitate this, ConfinedStreamingMethod must flag the particles which were bounced back from surface.
+ */
+
+class PYBIND11_EXPORT DryingDropletStreamingMethod : public mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param R is the radius of sphere
+         * \param bc is boundary conditions(slip or no-slip)
+         */
+        DryingDropletStreamingMethod(std::shared_ptr<mpcd::SystemData> sysdata,
+                                    unsigned int cur_timestep,
+                                    unsigned int period,
+                                    int phase, std::shared_ptr<::Variant> R, const Scalar density, unsigned int seed, mpcd::detail::boundary bc)
+        : mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>(sysdata, cur_timestep, period, phase, std::shared_ptr<mpcd::detail::SphereGeometry>()),
+          m_R(R), m_density(density), m_seed(seed), m_picks(m_exec_conf), m_bc(bc)
+          {}
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+
+    protected:
+        std::shared_ptr<::Variant> m_R;      //!< Radius of Sphere
+        const Scalar m_density;              //!< Solvent density
+        unsigned int m_seed;                 //!< Seed to evaporator pseudo-random number generator
+        const mpcd::detail::boundary m_bc;   //!< Boundary condition
+        unsigned int m_Npick;                              //!< Number of particles picked for evaporation on this rank
+
+        GPUArray<unsigned int> m_bounced_index;           //!< Indices of bounced particles
+        GPUVector<unsigned int> m_picks;                   //!< Particles picked for evaporation on this rank
+
+        virtual void applyPicks();                          //!< Apply the picks
+
+    private:
+        std::vector<unsigned int> m_all_picks;             //!< All picked particles on all the ranks
+
+        //!< For Making a random pick of particles across all ranks
+        void makeAllPicks(unsigned int timestep, unsigned int N_pick, unsigned int N_bounced_total);
+
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+
+void DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    /*
+     * Stream will only happen at streaming period and
+     * geometry will only be updated when stream will happen
+     */
+    if(timestep%m_period != 0) return;
```

Comment:
There is a function that will do this already and more robustly:
```suggestion
    // use peekStream since shouldStream will be called by parent class
    if(!peekStream(timestep)) return;
```